src/test/java/ru/hogwarts/school/controller/FacultyControllerMVCTest.java

```diff
@@ -0,0 +1,105 @@
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repository.FacultyRepository;
+import ru.hogwarts.school.repository.StudentRepository;
+import ru.hogwarts.school.service.StudentService;
+
+import java.util.List;
+import java.util.Optional;
+
+@WebMvcTest(controllers = FacultyController.class)
+public class FacultyControllerMVCTest {
+    @MockBean
+    StudentRepository studentRepository;
+    @MockBean
+    FacultyRepository facultyRepository;
+    @SpyBean
+    StudentService studentService;
+
+    @Autowired
+    MockMvc mockMvc;
+    ObjectMapper mapper = new ObjectMapper();
+
+    Optional<Faculty> optionalFaculty;
+    private Faculty faculty;
+
+    @BeforeEach
+    void init() {
+        faculty = new Faculty(1L, "stud_name", "black");
+        faculty.setStudents(List.of(new Student(1L, "stud_name", 21)));
+
+        optionalFaculty = Optional.of(faculty);
+        Mockito.when(facultyRepository.findById(ArgumentMatchers.anyLong()))
+                .thenReturn(optionalFaculty);
+        Mockito.when(facultyRepository.findAll())
+                .thenReturn(List.of(faculty));
+    }
+
+    @Test
+    void create() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/student")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(mapper.writeValueAsString(faculty)))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+        Mockito.verify(facultyRepository).save(ArgumentMatchers.any(Faculty.class));
+
+    }
+
+    @Test
+    void update() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.request(HttpMethod.PUT, "/student/1")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(mapper.writeValueAsString(faculty)))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+        Mockito.verify(facultyRepository).save(ArgumentMatchers.any(Faculty.class));
+
+    }
+
+    @Test
+    void delete() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.request(HttpMethod.DELETE, "/student/1")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(mapper.writeValueAsString(faculty)))
+                .andExpect(MockMvcResultMatchers.status().isOk());
+        Mockito.verify(facultyRepository).delete(ArgumentMatchers.any(Faculty.class));
+    }
+
+    @Test
+    void get() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.get("/student/1")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("id").isNotEmpty())
+                .andExpect(MockMvcResultMatchers.jsonPath("name").value("stud_name"))
+                .andExpect(MockMvcResultMatchers.jsonPath("age").value("20"));
```

Comment:
```suggestion
                .andExpect(MockMvcResultMatchers.jsonPath("color").value("black"));
```
У факультета цвет, а не возраст